ateam_geometry/test/variant_do_intersect_test.cpp

```diff
@@ -23,12 +23,37 @@
 #include "ateam_geometry/types.hpp"
 #include "ateam_geometry/variant_do_intersect.hpp"
 
-TEST(VariantDoIntersect, CircleToAnyIntersection)
+TEST(VariantDoIntersect, CircleToPoint)
 {
   ateam_geometry::Circle a(ateam_geometry::Point(0, 0), 1);
-  ateam_geometry::AnyShape b = ateam_geometry::Circle(ateam_geometry::Point(1, 0), 1);
+  ateam_geometry::AnyShape b = ateam_geometry::Point(0.5, 0.5);
   EXPECT_TRUE(ateam_geometry::variantDoIntersect(a, b));
 
-  a = ateam_geometry::Circle(ateam_geometry::Point(-2, 0), 1);
+  b = ateam_geometry::Point(1.5, 1.5);
   EXPECT_FALSE(ateam_geometry::variantDoIntersect(a, b));
 }
+
+TEST(VariantDoIntersect, CircleToCircle)
+{
+  ateam_geometry::Circle a(ateam_geometry::Point(0, 0), 1);
+  ateam_geometry::AnyShape b = ateam_geometry::Circle(ateam_geometry::Point(0, 0), 0.5);
+  EXPECT_TRUE(ateam_geometry::variantDoIntersect(a, b));
+
+  a = ateam_geometry::Circle(ateam_geometry::Point(0, 0), 1);
+  b = ateam_geometry::Circle(ateam_geometry::Point(0, 0), 1.5);
+  EXPECT_TRUE(ateam_geometry::variantDoIntersect(a, b));
+
+  a = ateam_geometry::Circle(ateam_geometry::Point(0, 0), 1);
+  b = ateam_geometry::Circle(ateam_geometry::Point(0, 0), 1);
+  EXPECT_TRUE(ateam_geometry::variantDoIntersect(a, b));
+}
+
+TEST(VariantDoIntersect, CircleToSegment)
+{
+  ateam_geometry::Circle a(ateam_geometry::Point(0, 0), 1);
+  ateam_geometry::AnyShape b = ateam_geometry::Segment(
+    ateam_geometry::Point(0, 0),
+    ateam_geometry::Point(1, 0)
+  );
+  EXPECT_TRUE(ateam_geometry::variantDoIntersect(a, b));
+}
```

Comment:
We should probably have at least one test for each of our new functions and each should include a intersecting and non-intersecting example.

Reply:
There should be a positive and negative test for every variant now